### DISCUSSION TITLE
Is it possible to get the original http.Request in an interceptor?

### DISCUSSION THREAD

**Discussion by nomad-software** [connectrpc/connect-go#457] [General]
Is it possible to get the original http.Request in an interceptor?

```
func MyInterceptor() connect.UnaryInterceptorFunc {
	fn := func(next connect.UnaryFunc) connect.UnaryFunc {
		return connect.UnaryFunc(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {

			r := req.HttpRequest() ? ? ?

			return next(ctx, req)
		})
	}
	return connect.UnaryInterceptorFunc(fn)
}
```

**Comment by akshayjshah** [comment]
No, it's not. What are you trying to accomplish?

Rather than using a Connect interceptor, you can access the request and response writer by building `net/http` middleware - all the generated server-side constructors return plain `http.Handler`s, which you can wrap with any logic you like.

**Comment by nomad-software** [comment]
I'm not really trying to accomplish anything, I just wondered. Thanks for the reply.

**Reply by akshayjshah** [reply]
:+1: No problem! I'm open to exposing more of the `*http.Request`, but in particular touching the body is very error-prone because `io.Reader`s are stateful.
